sound/soc/sof/sof-priv.h

```diff
@@ -245,6 +245,14 @@ struct snd_sof_dsp_ops {
 	/* pcm ack */
 	int (*pcm_ack)(struct snd_sof_dev *sdev, struct snd_pcm_substream *substream); /* optional */
 
+	/*
+	 * optional way to get output dma position for current stream in audio dsp
+	 * in some cases the position can't be gotten from shared memory
+	 */
```

Comment:
suggested edit: Optional callback to retrieve the link DMA position for the substream when the position is not reported in the shared SRAM windows but instead from a host-accessible hardware counter.

Reply:
sure, thanks